jar-infer/jar-infer-cli/build.gradle

```diff
@@ -39,12 +40,53 @@ shadowJar {
 shadowJar.dependsOn jar
 assemble.dependsOn shadowJar
 
-apply plugin: 'com.vanniktech.maven.publish'
+tasks.withType(PublishToMavenRepository) {
+    onlyIf {
+        publication == publishing.publications.shadow
+    }
+}
+tasks.withType(PublishToMavenLocal) {
+    onlyIf {
+        publication == publishing.publications.shadow
+    }
+}
 
 publishing {
     publications {
         shadow(MavenPublication) { publication ->
             project.shadow.component(publication)
+            afterEvaluate {
```

Comment:
Similarly, would be good to add a comment here, indicating that in the shadow configuration, we need to manually re-add the sources and javadoc jars and configure the pom, to pass the Maven Central checks